book/image-controller.md

```diff
@@ -1,2 +1,125 @@
 # Image Controller
 
+## Overview
+
+The Image Controller for AppStudio helps set up container image repositories
+for AppStudio Components. It watches Components to take actions according to
+the deletion event and the value set to specific annotations.
+
+External services are able to interact with image controller via these two
+annotations:
+
+- `image.redhat.com/generate`: if set, image controller starts to set up image
+  repository on Quay.io for a Component, or attempts to switch repository
+  visibility if the repository exists already.
+
+- `image.redhat.com/image`: information of created resources is set to this
+  annotation as a JSON string. Error message will also be set to here if there
+  is.
+
+## Set up image repository
+
+To request the controller to setup an image repository, annotate the Component
+CR with `image.redhat.com/generate: '{"visibility": "public"}'` or
```

Comment:
Can you define CR on its first use? Is there a reference to the Component CR(D) that you can link to?

Reply:
I think it could be https://redhat-appstudio.github.io/book/ref/application-environment-api.html#component, or link to the https://github.com/redhat-appstudio/application-api/blob/main/api/v1alpha1/component_types.go#L75 ?

image-controller just cares about the annotation, not the CR definition. I use term "Component CR" in order to keep consistency with other documents, like build-service.md.

 > Can you define CR on its first use?

image-controller starts to handle only when both `image.redhat.com/generate` and `component.Status.Devfile` are set.

Reply:
By defining CR, I just meant defining it as Custom Resource. It is a nit, I haven't looked to see if other documents define it.

Are those two references in sync with each other? If you refer to something, it would be better to refer to another place in the book over the code.

Reply:
> Are those two references in sync with each other? If you refer to something, it would be better to refer to another place in the book over the code.

I was told the API pages https://redhat-appstudio.github.io/book/ref/index.html are generated from the types defined in Go. I don't know if there is a mechanism to sync the possible changes regularly. If no, probably this is a point that can be improved, but the API should seldom change.